server/routes/bookVisit/chooseVisitTimeController.ts

```diff
@@ -61,8 +61,14 @@ export default class ChooseVisitTimeController {
       const selectedSessionTemplateReference = visitSessionSplit[1]
 
       const { booker, bookingJourney } = req.session
-      bookingJourney.selectedSessionDate = selectedSessionDate
-      bookingJourney.selectedSessionTemplateReference = selectedSessionTemplateReference
+
+      const selectedVisitSession = bookingJourney.allVisitSessions.find(
+        singleVisitSession =>
+          singleVisitSession.sessionTemplateReference.includes(selectedSessionTemplateReference) &&
+          singleVisitSession.sessionDate.includes(selectedSessionDate),
+      )
```

Comment:
```suggestion

      const selectedVisitSession = bookingJourney.allVisitSessions.find(
        session =>
          session.sessionTemplateReference === selectedSessionTemplateReference &&
          session.sessionDate === selectedSessionDate,
      )
```

We're just comparing strings here so no need for `.includes()` (which is for checking if a value is in an array)